### ISSUE TITLE
ListView FilterBar text color and background color not changing

### ISSUE THREAD

**Issue by preetvadaliya** [mit-cml/appinventor-sources#2612]
**Describe the bug**
From the community [post 1](https://community.appinventor.mit.edu/t/listview-search-bar-text-color-not-changing/46368), [post 2](https://community.appinventor.mit.edu/t/nb188-listview-filter-bar-issue/46035) the text color and background color are not changing according to device theme also there is no property for changing the text color and the background color of the filter bar.

**Affects**
- [x] Designer
- [ ] Blocks editor
- [x] Companion
- [x] Compiled apps
- [ ] Buildserver
- [ ] Debugging
- [ ] Other... (please describe)

**Expected behavior**
Text color and background color must be changed according to user choice or device theme.

**Steps to reproduce**
1. Create a new project 
2. Drag and drop ListView component from the component palette and enable the "ShowFilterBar" property
3. Change the theme from screen properties and check the appearance behavior of the filter bar on the device

**Comment by preetvadaliya** [comment]
ListView is one of the most important of all visible components, and it needs to add a few more properties to increase the user experience. I've created a list of new properties that we can add to enhance the user experience, here is the list...

1. BackGroundColor property for FilterBar
2. TextColor property for FilterBar
3. HintText property for FilterBar (user defined HintText for FilterBar)
4. HintTextColor property for FilterBar
5. TextSize property for FilterBar
6. FontTypeface property for FilterBar (for text and hint text both)
7. ImageShape property for ListView (default, rounded, rectangular, oval)
8. LineSeparator property for ListView #2597
9. TextAlignment property for ListView #2594 
10. Gesture Handling Events like swipe left & swipe right

@ewpatton @SusanRatiLane Can you both give me your opinion on this?

**Comment by halatmit** [comment]
Is there a design doc for this?  We need to keep app inventor accessible to
beginners.

**Comment by preetvadaliya** [comment]
@halatmit sir, no there is not any design doc for this because in the morning I found this issue after then I was looking at the code of ListView and I realize that these properties also need to add to ListView.

**Comment by halatmit** [comment]
We need to make sure that this long list of properties does not become
overwhelming.  It's worth thinking about the UI here.

**Comment by preetvadaliya** [comment]
@halatmit sir, as your suggestion I have created this small design doc here is the link 

https://docs.google.com/document/d/1x5SD1uo5gVJ92Q6x-wuFR4Q9UQ6Sugzo1hnjKWSQqrE/edit?usp=sharing

**Comment by halatmit** [comment]
Thanks for getting started on this,   It illustrates the result of
selecting different options.   But the design also
needs to indicate how users select these.   Are you planning a separate
property block for each of these with its own setter and getter?
That might be overwhelming for beginning users.  is there another way to do
this?



== Hal

Hal Abelson
***@***.***
Prof. of Comp. Sci. and Eng.
MIT Dept. of Elec. Eng. and Comp. Sci.

**Comment by preetvadaliya** [comment]
> Are you planning a separate property block for each of these with its own setter and getter?

yes, but some of them are only designer property like we have FontTypeface is designer only in Label and TextBox components.

> is there another way to do this?

I think there is no other way to do this but instead of making designer properties, we can simply add them as block properties so if the user wants to change any of them then they can change or update by calling the setter blocks.

**Comment by halatmit** [comment]
It would be bad to have the number of blocks keep increasing and
increasing.   That would make it overwhelming for beginners.
We need to come up with a more usable design. See what you can suggest.

== Hal

Hal Abelson
***@***.***
Prof. of Comp. Sci. and Eng.
MIT Dept. of Elec. Eng. and Comp. Sci.

**Comment by preetvadaliya** [comment]
At this moment I can't able to find any other usable design, but I think ListView is a collection of components like there are TextBox for FilterBar, two Labels(textviews) for main text and detail text, so indirectly we can say that the ListView is 4 or 5 visible components so definitely it have more properties compare to others.

**Comment by 06022000** [comment]
i can do it but i am not getting the file from where i should edit the code can you help me with that if possible

**Comment by preetvadaliya** [comment]
@06022000 I am already working on this but there is also some improvement need in filtering.

Searching operation become to slow when there is thousands of item in list.

**Comment by SusanRatiLane** [comment]
@preetvadaliya I don't think we want individual properties for the filter bar at this time. We do want the filter bar to visible in all themes, which is apparently an issue for one theme.

Filter performance is a much higher priority.

**Comment by halatmit** [comment]
Susan,

Let's chat about this.

== Hal

Hal Abelson
***@***.***
Prof. of Comp. Sci. and Eng.
MIT Dept. of Elec. Eng. and Comp. Sci.

**Comment by preetvadaliya** [comment]
> We do want the filter bar to visible in all themes, which is apparently an issue for one theme.

note that the back-ground color of the filter bar is also not change when we change the back-ground color of the list view and it looks wired ugly.

**Comment by SusanRatiLane** [comment]
If the filter bar was visible in all themes in the previous single-text version of ListView, my first priority would be to make the behavior of the filter bar match that version. Anything beyond that we consider an enhancement. There are some good enhancement ideas here, but they need to be balanced against making the component more complicated and confusing. I'll discuss with Hal and the rest of the team.
